utilities/PerformanceTesting/GetAverages.py

```diff
@@ -0,0 +1,181 @@
+import argparse
+import copy
+import json
+import sys
+from argparse import RawTextHelpFormatter
+from statistics import mean
+
+
+def get_file_paths(file_prefix, num_files):
+    """
+    Generates the json file paths of the format <file_prefix>idx.json
+    where idx is an integer between 0 and num_files-1
+
+    :param file_prefix: The first part of the file path
+    that all files to be averaged have in common
+    :param num_files: The number of files to be averaged
+
+    :return: A list of files to be averaged
+    """
+    files = []
+
+    for i in range(num_files):
+        files.append(f"{file_prefix}{i}.json")
+
+    return files
+
+
+def get_metricss(file_paths):
+    """
+    Reads the json files to be averaged and returns the json contents
+
+    :param file_paths: A list of file paths
+
+    :return: A list of dictionaries, which represent the metrics obtained
+    from one run of Collector
+    """
+    metricss = []
+
+    for file_path in file_paths:
+        with open(file_path) as f:
+            metricss.append(json.load(f))
+
+    return metricss
+
+
+def reset_cumulative(cumulative):
+    """
+    Loops through cumulative and sets the value of any entry with the key
+    'value' to an empty list
+
+    :param cumulative: A dictionary which will contain results from all runs
+    of the performance tests for a particular version of Collector later in
+    the program. Here it will only contain empty lists.
+    """
+    for key in cumulative:
+        if 'value' in cumulative[key]:
+            cumulative[key]['value'] = []
+        else:
+            reset_cumulative(cumulative[key])
+
+
+def accumulate_metric_values_once(metrics, cumulative):
+    """
+    Loops through the dictionary metrics and appends any values it find to
+    corresponding lists in the dictionary cumulative
+
+    :param metrics: A dictionary representing the results of one run of StackRox
+    :param cumulative: A dictionary which will contain results from all runs
+    of the performance tests for a particular version of Collector
+    """
+    for key in metrics:
+        if 'value' in metrics[key]:
+            if metrics[key]['value'] is not None:
+                cumulative[key]['value'].append(metrics[key]['value'])
+        else:
+            accumulate_metric_values_once(metrics[key], cumulative[key])
```

Comment:
The complexity of the output file format has become more complex over time. In the beginning the output was just
#############
metric_group_a
metric_a1: value_a1
metric_a2: value_a2
#############
metric_group_b
metric_b1: value_b1
metric_b2: value_b2
#############

Giles suggested changing the format to json. I then changed the output to 
{
    "metric_group_a": {
        "metric_a1": {"value": value_a1}
...

The reason for having 

"metric_a1": {"value": value_a1} 

instead of just simply

"metric_a1": value_a1

Is that I was thinking about adding other fields such as standard deviation in the future

"metric_a1": {"value": value_a1, "standard_deviation": standard_deviation_a1}

You suggested having something like this 

{
    "metric_group_a": {
        "metric_a1": {"
            "Average": {"value": ave_value_a1},
            "Maximum": {"value": max_value_a1}
        }
...
 
However not all metrics have an average and maximum, so we don't know how far deep we need to go to find the value. That is where recursive depth first search comes in handy.